Remove Python 3.6 from tests to fix tests

RytoEX opened the pull request:
Python 3.6 is not available on ubuntu-latest (22.04), so the job fails when it cannot find Python 3.6. Python 3.6 was available when ubuntu-latest was 20.04, which is why the test used to work.

Alternatively, you could explicitly set ubuntu-20.04 and/or ubuntu-22.04 in the OS matrix. If you wanted both, you'd have to do some checking to ensure you're not invoking an invalid combination such as ubuntu-22.04 and Python 3.6.

marian-code commented:
Great, thanks for the fix, unfortunatelly I do not have much times these days to maintain this.
Since we are doing this I guess it makes sense to add python 3.11 and 3.12. Would you mind doing that?

RytoEX commented:
> Great, thanks for the fix, unfortunatelly I do not have much times these days to maintain this. Since we are doing this I guess it makes sense to add python 3.11 and 3.12. Would you mind doing that?

Sure. I can add them in this PR or in a separate PR, whichever you prefer.

marian-code commented:
add them to this PR please

RytoEX commented:
> add them to this PR please

Added as a second commit.

RytoEX commented:
Hmm. Running the Python 3.11 tests now on my fork, Python 3.11 fails with some error, so maybe adding Python 3.11 and 3.12 should be addressed in a separate PR pending some investigation.

RytoEX commented:
> Hmm. Running the Python 3.11 tests now on my fork, Python 3.11 fails with some error, so maybe adding Python 3.11 and 3.12 should be addressed in a separate PR pending some investigation.

I think I have a lead on why Python 3.11 and 3.12 are failing (seems to be outdated pylint requiring astroid requiring outdated wrapt), but I'd like to revert this to just removing Python 3.6 and tackle Python 3.11 and 3.12 in a separate PR, if that's okay.

marian-code commented:
Sure, go ahead, I looked it up on my end too and it seems to be caused just as you said by the fact that the dependencies are pinned to fixed versions which aparently do not have support for python>3.10

RytoEX committed:
Remove Python 3.6 from tests to fix tests

Python 3.6 is not available on ubuntu-latest (22.04).
It was available when ubuntu-latest was 20.04.

RytoEX commented:
Reverted back to just the Python 3.6 removal.

> Sure, go ahead, I looked it up on my end too and it seems to be caused just as you said by the fact that the dependencies are pinned to fixed versions which aparently do not have support for python>3.10

It seems that pylint 3 only supports Python 3.8+, while pylint 2.17 still supports Python 3.7. ~~I'm not sure which version of pylint has the correct dependency chain that works with Python 3.11+.~~ Seemingly, it would be pylint 2.15 that is needed.

Python 3.7 has also reached end-of-life, so there is the option to also drop Python 3.7 and support only 3.8+. I leave which versions to support up to you. If it is desired to also drop 3.7, I can address that here or as a separate PR (my preference would be a separate PR, but up to you how you'd like to handle it, if at all).

marian-code commented:
Ok let s drop 3.8 in next PR
